<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1376</width>
    <height>916</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QGroupBox" name="settingsBox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>220</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>220</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>Settings</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QLabel" name="selectedLabel">
         <property name="text">
          <string>Selected Object: </string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="selectedObjectLabel">
         <property name="text">
          <string>Nothing selected..</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="verticalGroupBox">
         <property name="title">
          <string>Animation</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_9">
          <item>
           <widget class="QCheckBox" name="animationCheckBox">
            <property name="text">
             <string>Enabled</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="resetAnimationButton">
            <property name="text">
             <string>Reset</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="rotationBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>205</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>205</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="title">
          <string>Rotation</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QDial" name="RotationDialX">
              <property name="toolTip">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Rotation along the X-axis&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="maximum">
               <number>359</number>
              </property>
              <property name="wrapping">
               <bool>true</bool>
              </property>
              <property name="notchTarget">
               <double>12.000000000000000</double>
              </property>
              <property name="notchesVisible">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDial" name="RotationDialY">
              <property name="toolTip">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Rotation along the Y-axis&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="maximum">
               <number>359</number>
              </property>
              <property name="wrapping">
               <bool>true</bool>
              </property>
              <property name="notchTarget">
               <double>12.000000000000000</double>
              </property>
              <property name="notchesVisible">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDial" name="RotationDialZ">
              <property name="toolTip">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Rotation along the Z-axis&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="maximum">
               <number>359</number>
              </property>
              <property name="wrapping">
               <bool>true</bool>
              </property>
              <property name="notchTarget">
               <double>12.000000000000000</double>
              </property>
              <property name="notchesVisible">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QPushButton" name="ResetRotationButton">
            <property name="text">
             <string>Reset Rotation</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="scalingBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>205</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>205</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="title">
          <string>Scaling</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QSlider" name="ScaleSlider">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>600</number>
            </property>
            <property name="value">
             <number>100</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="invertedAppearance">
             <bool>false</bool>
            </property>
            <property name="invertedControls">
             <bool>false</bool>
            </property>
            <property name="tickPosition">
             <enum>QSlider::NoTicks</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="ResetScaleButton">
            <property name="text">
             <string>Reset Scaling</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="shadingBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>205</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>205</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="title">
          <string>Shading</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QRadioButton" name="PhongButton">
            <property name="text">
             <string>Pho&amp;ng</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="NormalButton">
            <property name="text">
             <string>Nor&amp;mal</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="GouraudButton">
            <property name="text">
             <string>&amp;Gouraud</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="WaterButton">
            <property name="text">
             <string>Water</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="realismButton">
            <property name="text">
             <string>Metallic</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="verticalGroupBox">
         <property name="title">
          <string>Texturing</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <widget class="QCheckBox" name="checkBox">
            <property name="text">
             <string>Wireframe</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="showFiltersButton">
            <property name="text">
             <string>Show Filter Options</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="moreOptionsButton">
         <property name="text">
          <string>More Options..</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="filterGroupBox">
      <property name="maximumSize">
       <size>
        <width>300</width>
        <height>16777215</height>
       </size>
      </property>
      <layout class="QVBoxLayout" name="filterOptionsGroupBox">
       <item>
        <widget class="QCheckBox" name="anisotropicFilter">
         <property name="text">
          <string>Anisotropic  Filtering</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="wrappingBox">
         <property name="title">
          <string>Wrapping Method</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <widget class="QRadioButton" name="wrapRepeat">
            <property name="text">
             <string>GL_REPEAT</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="wrapMirror">
            <property name="text">
             <string>GL_MIRRORED_REPEAT</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="wrapEdgeClamp">
            <property name="text">
             <string>GL_CLAMP_TO_EDGE</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="wrapBorderClamp">
            <property name="text">
             <string>GL_CLAMP_TO_BORDER</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="maxFilterBox">
         <property name="title">
          <string>Magnifying Method</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_8">
          <item>
           <widget class="QRadioButton" name="maxNearest">
            <property name="text">
             <string>GL_NEAREST</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="maxLinear">
            <property name="text">
             <string>GL_LINEAR</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="minFilterBox">
         <property name="title">
          <string>Minifying Method</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <widget class="QRadioButton" name="minNearest">
            <property name="text">
             <string>GL_NEAREST</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="minLinear">
            <property name="text">
             <string>GL_LINEAR</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="minMipNearestNearest">
            <property name="text">
             <string>GL_NEAREST_MIPMAP_NEAREST</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="minMipNearestLinear">
            <property name="text">
             <string>GL_NEAREST_MIPMAP_LINEAR</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="minMipLinearNearest">
            <property name="text">
             <string>GL_LINEAR_MIPMAP_NEAREST</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="minMipLinearLinear">
            <property name="text">
             <string>GL_LINEAR_MIPMAP_LINEAR</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="MainView" name="mainView">
      <property name="minimumSize">
       <size>
        <width>580</width>
        <height>0</height>
       </size>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="settingsBox2">
      <property name="maximumSize">
       <size>
        <width>240</width>
        <height>16777215</height>
       </size>
      </property>
      <layout class="QVBoxLayout" name="shadingSettings">
       <item>
        <widget class="QGroupBox" name="lightingBox">
         <property name="title">
          <string>Lighting</string>
         </property>
         <layout class="QVBoxLayout" name="lightBox">
          <item>
           <widget class="QLabel" name="ambientLabel">
            <property name="text">
             <string>Ambient</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="AmbientSlider">
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="value">
             <number>30</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="diffuseLabel">
            <property name="text">
             <string>Diffuse</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="DiffuseSlider">
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="value">
             <number>70</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="specularLabel">
            <property name="text">
             <string>Specular</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="SpecularSlider">
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="value">
             <number>99</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="specularExpLabel">
            <property name="text">
             <string>Specular Exponent</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="SpecExpSlider">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>256</number>
            </property>
            <property name="value">
             <number>32</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="lightColorBox">
            <property name="title">
             <string>Light Color</string>
            </property>
            <layout class="QHBoxLayout" name="LightColorBox">
             <item>
              <widget class="QSpinBox" name="red">
               <property name="maximum">
                <number>255</number>
               </property>
               <property name="value">
                <number>255</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="green">
               <property name="maximum">
                <number>255</number>
               </property>
               <property name="value">
                <number>255</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="blue">
               <property name="maximum">
                <number>255</number>
               </property>
               <property name="value">
                <number>255</number>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="lightPosLabel">
            <property name="text">
             <string>Light Position</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="lightPosLabelBox">
            <item>
             <widget class="QLabel" name="xLabel">
              <property name="minimumSize">
               <size>
                <width>12</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>8</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>x</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="yLabel">
              <property name="maximumSize">
               <size>
                <width>12</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>y</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="zLabel">
              <property name="maximumSize">
               <size>
                <width>12</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>z</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="lightPosBox">
            <item>
             <widget class="QSlider" name="XSLider">
              <property name="minimum">
               <number>-100</number>
              </property>
              <property name="maximum">
               <number>100</number>
              </property>
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSlider" name="YSLider">
              <property name="minimum">
               <number>-100</number>
              </property>
              <property name="maximum">
               <number>100</number>
              </property>
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSlider" name="ZSLider">
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>50</height>
               </size>
              </property>
              <property name="minimum">
               <number>-100</number>
              </property>
              <property name="maximum">
               <number>100</number>
              </property>
              <property name="value">
               <number>100</number>
              </property>
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="materialBox">
         <property name="title">
          <string>Material</string>
         </property>
         <layout class="QVBoxLayout" name="box">
          <item>
           <widget class="QCheckBox" name="DiffuseTex">
            <property name="text">
             <string>Diffuse Texture</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="SpecularTex">
            <property name="text">
             <string>Specular Map</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="NormalTex">
            <property name="text">
             <string>Normal Map</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="multiplierBox">
            <layout class="QVBoxLayout" name="normalIntensityBox">
             <item>
              <widget class="QLabel" name="specularMultLabel">
               <property name="text">
                <string>Specular Multiplier</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="SpecularMultiplier">
               <property name="value">
                <double>1.000000000000000</double>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="normalMultLabel">
               <property name="text">
                <string>Normal Multiplier</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="NormalMultiplier">
               <property name="minimum">
                <double>0.000000000000000</double>
               </property>
               <property name="maximum">
                <double>5.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.100000000000000</double>
               </property>
               <property name="value">
                <double>1.000000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MainView</class>
   <extends>QOpenGLWidget</extends>
   <header location="global">mainview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
